packages/artifact/__tests__/download.test.ts

```diff
@@ -188,6 +188,53 @@ describe('Download Tests', () => {
     }
   })
 
+  it('Test retry on truncated response with gzip', async () => {
+    const fileContents = Buffer.from(
+      'Sometimes gunzip fails on the first try\n',
+      defaultEncoding
+    )
+    const targetPath = path.join(root, 'FileD.txt')
+
+    setupDownloadItemResponse(true, 200, true, fileContents, true)
+    setupThrowWhenReadingStream()
```

Comment:
I don't think this `spyon` call is necessary. When you gzip the `fileContents` buffer and truncate it, the gunzip stream in `pipeResponseToFile` should raise an error.

Reply:
Hmm, from my local testing, I needed this.  We do not raise any error automatically from `pipeResponseToFile`.  If I comment out this `setupThrowWhenReadingStream()` the test will fail because we successfully downloaded the truncated file and didn't retry.

Reply:
Odd, I try commenting it out and it passed for me. https://github.com/hashtagchris/toolkit/commit/79dfd4df56ec318d09c32231df11601f8707dd04

Reply:
Hmm you are right!  I think something was cached in my terminal so I wasn't running with the latest code?  Weird but in a new terminal I was able to remove this line and everything worked.